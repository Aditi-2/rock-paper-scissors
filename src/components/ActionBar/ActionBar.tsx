import styled from "@emotion/styled";
import React from "react";
import { ButtonType } from "../../common/types/game";
import { CustomButton } from "../CustomButton/CustomButton";

interface ActionBarProps {
  onButtonClick: (buttonType: ButtonType) => void | Promise<void>;
}

export const ActionBar: React.FC<ActionBarProps> = ({ onButtonClick }) => (
  <ActionButtonContainer>
    <CustomButton onClick={() => onButtonClick("RESTART")} label="Play again" />
    <CustomButton onClick={() => onButtonClick("END")} label="End game" />
  </ActionButtonContainer>
);

const ActionButtonContainer = styled.div`
  grid-area: action-buttons;
  margin-top: 3rem;
`;