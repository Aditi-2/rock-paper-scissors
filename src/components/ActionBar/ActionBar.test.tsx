import React from "react";
import { create } from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";
import { ActionBar } from "./ActionBar";

const onClick = jest.fn();

const getComponent = () => <ActionBar onButtonClick={onClick} />;

test("ActionBar Component snapshot", () => {
  const component = create(getComponent()).toJSON();
  expect(component).toMatchSnapshot();
});

test("renders ActionBar Component", () => {
  render(getComponent());
  const buttonOne = screen.getByText(/Play again/i);
  expect(buttonOne).toBeInTheDocument();
  const buttonTwo = screen.getByText(/End game/i);
  expect(buttonTwo).toBeInTheDocument();
});

test("ActionBar Component onClick is called", () => {
  render(getComponent());
  const buttonOne = screen.getByText(/Play again/i);
  fireEvent.click(buttonOne);
  expect(onClick).toHaveBeenCalledWith("RESTART");

  const buttonTwo = screen.getByText(/End game/i);
  fireEvent.click(buttonTwo);
  expect(onClick).toHaveBeenCalledWith("END");
});
