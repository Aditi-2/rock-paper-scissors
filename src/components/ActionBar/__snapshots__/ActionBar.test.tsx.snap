// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar Component snapshot 1`] = `
<div
  className="css-10obzey"
>
  <button
    className="custom-btn ripple css-zsodnc"
    onClick={[Function]}
  >
    Play again
  </button>
  <button
    className="custom-btn ripple css-zsodnc"
    onClick={[Function]}
  >
    End game
  </button>
</div>
`;
